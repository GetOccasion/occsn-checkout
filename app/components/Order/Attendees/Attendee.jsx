import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Field } from 'mitragyna';

import s from 'underscore.string';

import occsn from '../../../libs/Occasion';

import { Card, CardBody, CardTitle, CardText, FormGroup, Input } from 'reactstrap';

export default class Attendee extends PureComponent {
  static propTypes = {
    indexOf: PropTypes.number.isRequired,
    questions: PropTypes.shape({
      __collection: PropTypes.arrayOf(PropTypes.string)
    }).isRequired,
    subject: PropTypes.instanceOf(occsn.Attendee).isRequired,
  };

  render() {
    let { indexOf, questions } = this.props;

    return <Card className="attendee-container">
      <CardBody className="attendee">
        <CardTitle className="attendee-title">
          Attendee { indexOf + 1 }
        </CardTitle>
        {
          questions.map((q) => {
            return <FormGroup className="attendee-input-container">
              <Field type='text' name={q} placeholder={s.humanize(q)} component={Input} invalidClassName='is-invalid'></Field>
            </FormGroup>;
          }).toArray()
        }
      </CardBody>
    </Card>;
  }
}
