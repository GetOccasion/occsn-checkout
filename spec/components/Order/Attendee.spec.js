import React from 'react';
import PropTypes from 'prop-types';

import Attendee from 'app/components/Order/Attendees/Attendee';

import axios from 'axios';
import occsn from 'app/libs/Occasion';

import productFixture from 'fixtures/products/attendee_questions.json';

describe('Order', () => {
  describe('Attendee', () => {
    let attendee, product, wrapper;

    async function setupWrapper(customResponses, customProps) {
      let responses = {
        '/products/:id/': { status: 200, data: productFixture },
        ...customResponses
      };
      axios._setMockResponses(responses);

      product = await occsn.Product.find('1');
      attendee = occsn.Order.build().attendees().build();

      let props = {
        indexOf: 1,
        questions: product.attendeeQuestions,
        subject: attendee,
        ...customProps,
      };

      let context = {
        resource: attendee
      };

      let childContextTypes = {
        resource: PropTypes.object
      };

      wrapper = shallow(<Attendee {...props} />, { context, childContextTypes });
    }

    beforeEach(async () => {
      await setupWrapper({}, {});
    });

    afterEach(async () => {
      axios.reset();
    });

    it('matches snapshot', () => {
      expect(wrapper.find('Card')).toMatchSnapshot();
    });
  });
});
