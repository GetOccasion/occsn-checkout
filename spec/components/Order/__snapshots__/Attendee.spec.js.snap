// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Attendee matches snapshot 1`] = `
<Card
  className="mb-2"
  tag="div"
>
  <CardBody
    tag="div"
  >
    <CardTitle
      tag="h5"
    >
      Attendee 
      1
    </CardTitle>
    <FormGroup
      tag="div"
    >
      <Field
        component={[Function]}
        invalidClassName="is-invalid"
        name="firstName"
        placeholder="First name"
        type="text"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Field
        component={[Function]}
        invalidClassName="is-invalid"
        name="lastName"
        placeholder="Last name"
        type="text"
      />
    </FormGroup>
  </CardBody>
</Card>
`;
